include/EntryPointConfirmOptIn.php

```diff
@@ -43,40 +43,104 @@
 /**
  * Class confirm_opt_in
  */
-class ConfirmOptIn
+class EntryPointConfirmOptIn
 {
     /**
      * @var EmailAddress $emailAddress
      */
     private $emailAddress;
+    
+    /**
+     * 
+     * @param array $request
+     * @param array $post
+     */
+    public function __construct($request = null, $post = null) {
+        
+        if(is_null($request)) {
+            $request = $_REQUEST;
+        }
+        
+        if(is_null($post)) {
+            $post = $_POST;
+        }
+        
+        $method = isset($request['method']) && $request['method'] ? $request['method'] : null;
+        
+        switch($method) {
+            case 'confirmOptInSelected':
+                $output = $this->methodConfirmOptInSelected($post);
+                break;
+            default:
+                $output = $this->methodConfirmOptInUser($request);
+                break;
+        }
+        
+        echo $output;
+        sugar_cleanup();
+    }
+    
+    /**
+     * 
+     * @global array $sugar_config
+     * @param array $post
+     * @return string|boolean
+     */
+    private function methodConfirmOptInSelected($post) {
+        
+        global $sugar_config;
+        
+        $confirmOptInEnabled = isset($sugar_config['email_enable_confirm_opt_in']) && $sugar_config['email_enable_confirm_opt_in'];
+
+        if (!$confirmOptInEnabled) {
+            $this->warn('Confirm Opt In disabled');
+            return false;
+        }
+
+        
+        $module = $post['module'];
+        $uids = explode(',', $post['uid']);
+        $emailMan = new EmailMan();
+        $err = 0;
+        $msg = '';
+        foreach($uids as $uid) {
+            if(!$emailMan->addOptInEmailToEmailQueue($module, $uid)) {
+                $err++;
+            }
+        }
+        
+        if($err) {
+            $msg = 'Incorrect Bean ID. ';
+        } else {
+            $msg = 'All ' . $module . ' added to email queue.';
+        }
+        
+        return $msg;
+    }
 
     /**
-     * Set up
+     * Confirm Opt In User
+     * 
+     * @param array $request
+     * @return string
      */
-    public function pre_display()
+    private function methodConfirmOptInUser($request)
```

Comment:
I am concerted that this will leave a nasty merge conflict in git.

In relation to my previous comment, I mentioned that this is a view. Expects a contact, account, lead, or target. So naming it with User at the end is unclear. The tick should not be displayed in users and employees.